File: projects/10/java/FlyingBird.java
/* FlyingBird.java provides subclass for a Bird class.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Sinai Park
 * Date: April 24 2020
 ******************************************************/


public class FlyingBird extends Bird {


    /* explicit constructor
    * Receive: name, a String 
    * PostCond: myName == name.
    */
    public FlyingBird(String name)
    {
    super(name);
    }
    
     /* A FlyingBird's Movement
      * Return: a flying-movement ("flew past!").
      */
    public String movement()
    {
        return "flew past";
    }
    
    }

File: projects/10/clojure/src/FlyingBird.clj
;;;; FlyingBird.clj provides a FlyingBird "class".
;;;; 
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by: Sinai Park
;;;; Date:April 24 2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

;; bring in 'Bird' (to be used as superclass) 
(load "Bird")

;; define 'Duck' as 'subclass' of Bird
(defrecord FlyingBird [^Bird name] )

;;; constructors
;;; - default: takes no arguments
;;;   Postcondition: name is initialized to a default value
;;; - explicit: takes one parameter
;;;   Receive: itsName, a String
;;;   Postcondition: name == itsName
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn make-FlyingBird
    ([]                (->FlyingBird "Ann Onymous"))
    ([^String itsName] (->FlyingBird itsName))
  )

;;; method to retrieve a Bird's Movement
;;; Receive: this, a FlyingBird's movement
;;; Return: "flew past"
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defmethod Movement FlyingBird [ _ ]
    "flew past"
  )